pkgs/test_core/lib/direct_run.dart

```diff
@@ -0,0 +1,41 @@
+import 'dart:async';
+
+import 'package:path/path.dart' as p;
+import 'package:test_api/backend.dart'; //ignore: deprecated_member_use
+import 'package:test_api/src/backend/declarer.dart'; //ignore: implementation_imports
+import 'package:test_api/src/backend/invoker.dart'; // ignore: implementation_imports
+import 'package:test_api/src/frontend/utils.dart'; // ignore: implementation_imports
+import 'package:test_api/src/utils.dart'; // ignore: implementation_imports
+
+import 'src/runner/configuration.dart';
+import 'src/runner/engine.dart';
+import 'src/runner/plugin/environment.dart';
+import 'src/runner/reporter.dart';
+import 'src/runner/reporter/expanded.dart';
+import 'src/runner/runner_suite.dart';
+import 'src/runner/suite.dart';
+import 'src/util/print_sink.dart';
+
+Future<bool> directRunTests(FutureOr<void> Function() testMain,
+    {Reporter Function(Engine)? reporter}) async {
```

Comment:
nit: the variable name should indicate this is really a factory, `reporterFactory`, `makeReporter`, `buildReporter`, etc

Reply:
Done